worker/src/Settings.cpp

```diff
@@ -62,6 +67,10 @@ void Settings::SetConfiguration(int argc, char* argv[])
 
 	/* Parse command line options. */
 
+	// getopt_long_only() is not thread-safe
+	std::lock_guard<std::mutex> lock(globalSyncMutex);
+
+	optind = 1; // Set explicitly, otherwise subsequent runs will fail
```

Comment:
Dot at the end of comments, please. Also, why this? Can you explain please?

```suggestion
	optind = 1; // Set explicitly, otherwise subsequent runs will fail.
```

Reply:
I was very surprised that `getopt_long_only()` uses those global variables.
`optind` is initialized as `1` by default and is incremented as `getopt_long_only()` iterates through `argv`. Since now we are expecting this function to be called multiple times we need to reset `optind` back to its default value before use or it doesn't work properly.

I really truly do not understand why this trivial function needs global state, but hey, it does.